src/client/src/rt-components/Environment/EnvironmentContext.tsx

```diff
@@ -1,19 +1,39 @@
 import React from 'react'
 
-export interface EnvironmentValue {
-  isDesktop: Boolean
-  openfin?: Window
+export interface EnvironmentValue<Provider extends Window = Window> {
+  provider: Provider
+  openfin?: Provider | null
```

Comment:
openFin will be of type OpenFin

Reply:
That's a matter for the end user as the Environment does not currently own knowledge about the provider other than a duck type of what must be satisfied.